faker/providers/misc/__init__.py

```diff
@@ -130,11 +130,14 @@ def locale(self):
     def language_code(self):
         return self.random_element(Provider.language_locale_codes.keys())
 
-    def uuid4(self):
+    def uuid4(self, uuid_object=False):
```

Comment:
I would rather have a `cast_to` argument for more flexibility. Something like:

```python
def uuid(self, cast_to=str):
    value = uuid.uuid4()
    if cast_to is not None:
        return cast_to(value)
    return value
```

This would allow users to cast to `int`, or to `hex`:

```python
# currently UUIDs can't be casted with `hex`. See https://bugs.python.org/issue35115
from operator import attrgetter

faker.uuid4(cast_to=int)
faker.uuid4(cast_to=attrgetter('hex'))

Reply:
Good idea I'll start working on changing it